File: use-cases/src/main/java/com/octo/usecases/CityForecastPresenter.java
package com.octo.usecases;

public interface CityForecastPresenter {
    void onEmptyInput();

    void onUnknownCity();

    void onGenericException();

    void onUnavailableForecasts();

    void onForecasts(Forecast worstForecast, Forecast bestForecast);
}

File: use-cases/src/main/java/com/octo/usecases/CityForecastPresenter.kt
package com.octo.usecases

import com.octo.repository.network.Forecast

interface CityForecastPresenter {
    fun onEmptyInput()
    fun onGenericException()
    fun onUnavailableForecasts()
    fun onForecasts(worstForecast: Forecast?, bestForecast: Forecast?)
}